lib/archethic_web/controllers/faucet_controller.ex

```diff
@@ -0,0 +1,137 @@
+defmodule ArchEthicWeb.FaucetController do
+  @moduledoc false
+
+  alias ArchEthic.TransactionChain.{
+    Transaction,
+    TransactionData,
+    TransactionData.Ledger,
+    TransactionData.UCOLedger
+  }
+
+  alias ArchEthic.Crypto
+
+  use ArchEthicWeb, :controller
+
+  def index(conn, __params) do
+    conn
+    |> put_resp_header("cache-control", "no-cache, no-store, must-revalidate")
+    |> put_resp_header("pragma", "no-cache")
+    |> put_resp_header("expires", "0")
+    |> render("index.html", address: "")
+  end
+
+  def create_transfer(conn, %{"address" => address}) do
+    case transfer(address, :secp256r1) do
+      :ok ->
+        conn
+        |> put_flash(:info, "Transferred successfully")
+        |> redirect(to: "/faucet")
+
+      {:error, _} ->
+        conn
+        |> put_flash(:error, "Unable to transfer")
+        |> render("index.html", address: address)
+    end
+  end
+
+  def transfer(
+        address,
+        curve \\ Crypto.default_curve()
+      )
+      when is_bitstring(address) do
+    gen_seed =
```

Comment:
I think for this one you can leverage module attributes generated at the compilation time, unless you want to customize it at runtime

Reply:
For the curve or pool_gen_seed?

Reply:
For the genesis seed for testnet

Reply:
I want to know what is ultimately the default curve for our whole project? Is it **secp256r1** or ### ed25519? Because the currently the default value of **ARCHETHIC_TESTNET_GENESIS_ADDRESS** is generated using **secp256r1** whereas our default curve in the config is **ed25519**.

Reply:
You have the capability to determine which curve to use because each public key is prepended by two bytes: curve id and origin id.
Using Crypto module you can determine the curve and you can use this one in the derivation key algorithm .

Reply:
Oh yes I can do that as well thanks.

Reply:
To derive the public key from seed, we need to know the curve first right?

Reply:
Suppose I have the seed and address (of pool) both as environment variables. Is it possible to derive the curve from these two informations?

Reply:
Is it possible to determine the curve just from seed/address? Because the private key is not available in environment.

If I can get the curve I can use this :
```elixir
new(type, data, seed, index, curve \\ Crypto.default_curve())
```

If I can get the private key I can use this instead : 
```elixir
new_with_keys(type, data, previous_private_key, previous_public_key, next_public_key)
```

But without either of them I can't generate a new transaction.

Reply:
Is it possible to get the public key from the seed/address? Without the public key I can't determine the curve.

If I can get the curve I can use this :
```elixir
new(type, data, seed, index, curve \\ Crypto.default_curve())
```

If I can get the private key I can use this instead : 
```elixir
new_with_keys(type, data, previous_private_key, previous_public_key, next_public_key)
```

But without either of them I can't generate a new transaction.

Reply:
I don't think it's possible.
We should rethink the way to add the testnet seed into the conf.
We could imagine something like:

config/config.exs
```elixir
 config :archethic, ArchEthicWeb.FaucetController,
      seed: "3A7B579DBFB7CEBE26293850058F180A65D6A3D2F6964543F5EDE07BEB2EFDA4" |> Base.decode16!
```

config/dev.exs

```elixir
config  :archethic, ArchEthic.Bootstrap.NetworkInit, 
   genesis_pools: [
      # Address generated from the faucet seed
      %{ address: "00EC64107CA604A6B954037CFA91ED18315A77A94FBAFD91275CEE07FA45EAF893", amount: 10000000}
   ]
  
```

config/prod.exs

```elixir
config :archethic, ArchEthic.Bootstrap.NetworkInit, 
  genesis_pools: [
    ...
    ...,
   (if System.get_env("ARCHETHIC_TESTNET", "false") == "true", do:       
       "00EC64107CA604A6B954037CFA91ED18315A77A94FBAFD91275CEE07FA45EAF893"
    )
  ]
  |> Enum.filter(& &1)
```

Reply:
Suppose I have both the seed and address now. But I don't think I can do a new transaction without knowing the curve which I cannot derive without knowing the public key.

Reply:
You can because the address uses the default curve, as we are not customizing the faucet address

Reply:
But in docsify it is mentioned we can customise the pool address. Should we remove that from doc?

Reply:
Yes we will if it works more

Reply:
> You can because the address uses the default curve, as we are not customizing the faucet address

Right now the address is `0073bdaf847037115914ff5ca15e52d162db57b5089d5e4bf2005d825592c9c945` which has been derived using secp256r1. I am changing the existing pool address to the one using ed25519 default curve.